include/plugin/runtime-support.h

```diff
@@ -270,22 +274,54 @@ void confirm_protocol();
  */
 void reject_protocol(const std::string& reason);
 
+/**
+ * Opaque handle to a protocol analyzer.
+ */
+class ProtocolHandle {
```

Comment:
This isn't quite symmetric to how files do it: there we return a string. Maybe we should have originally introduced a `FileHandle` there, but I'm wondering if for symmetry using a string for protocols would be more consistent? But not fully sure what I like more.

Reply:
The issue with returning a `string` would be that we would need to parse it whenever we make use of the handle. I believe we should avoid that since it would add unnecessary cost to each call using a handle.